### ISSUE TITLE
Async _lastHidden capture: overlap with lm_head MatMul instead of pre-sync

### ISSUE THREAD

**Issue by pekkah** [pekkah/SharpInference#49]
## Background

In \`CudaHybridGdnForwardPass.Forward\` lines 1233-1242:

\`\`\`csharp
if (_hasMtp)
{
    _gpu.CopyDevice(_gpuLastHidden, _gpuHidden);
    _gpu.Download(_gpuLastHidden, new Span<float>(_lastHidden, _embDim));   // sync
}
_gpu.RmsNorm(_gpuHidden, _gpuHidden, _gpuOutputNorm!, _hp.RmsNormEps);
_gpu.MatMul(_gpuLogits, _gpuOutputWeight!, _gpuHidden, …);
_gpu.Download(_gpuLogits, _logitsBuf);                                       // sync
\`\`\`

The first \`_gpu.Download\` issues a \`cudaMemcpyAsync\` + \`StreamSynchronize\` (per \`DownloadViaStaging\`). The sync stalls the host before the \`lm_head\` MatMul can be enqueued — the GPU sits idle.

The host doesn't actually need \`_lastHidden\` until the MTP forward starts (which happens after the logits download completes), so the lm_head MatMul could be enqueued FIRST and the \`_lastHidden\` download could land in parallel on the same stream.

## Scope

Add a \`Download\` variant (or flag) that skips the immediate \`StreamSynchronize\`. Caller is responsible for ensuring the host buffer isn't read until a later sync point (which the \`_gpu.Download(_gpuLogits, _logitsBuf)\` call already provides — that one DOES sync, and by then \`_lastHidden\` will also have landed since both transfers go through the same stream).

\`\`\`csharp
_gpu.CopyDevice(_gpuLastHidden, _gpuHidden);
_gpu.DownloadAsync(_gpuLastHidden, new Span<float>(_lastHidden, _embDim));   // no sync
_gpu.RmsNorm(_gpuHidden, _gpuHidden, _gpuOutputNorm!, _hp.RmsNormEps);
_gpu.MatMul(_gpuLogits, _gpuOutputWeight!, _gpuHidden, …);
_gpu.Download(_gpuLogits, _logitsBuf);                                       // sync covers both
\`\`\`

Same change applies in \`BatchForward2\` at lines 1382-1383 (two LastHidden snapshots).

## Acceptance criteria

- [ ] Per-token decode improves on the 35B-A3B-MTP CUDA-hybrid bench by ~0.5 %.
- [ ] Verify ordering: \`_lastHidden\` host buffer is correct after \`_logitsBuf\` Download completes.
- [ ] No regression on dense 27B-MTP.

## Related

- #44 (surfaced this on the MTP path)
- The new pinned-buffers issue (independent; both shrink CUDA tail latency)
